common/flatpak-utils.c

```diff
@@ -105,12 +105,18 @@ flatpak_error_quark (void)
 gboolean
 flatpak_fail_error (GError **error, FlatpakError code, const char *fmt, ...)
 {
+  GError *new;
+
   if (error == NULL)
     return FALSE;
 
   va_list args;
   va_start (args, fmt);
-  GError *new = g_error_new_valist (FLATPAK_ERROR, code, fmt, args);
+  if (*error != NULL &&
+      g_error_matches (*error, G_IO_ERROR, G_IO_ERROR_CANCELLED))
```

Comment:
You don't need to check whether `*error` is `NULL` – `g_error_matches (NULL, ...)` is documented to be OK. But if `*error != NULL` then it is a programming error to call `g_propagate_error (error, ...)` below. So either callers are using `flatpak_fail` incorrectly, or all the changes in this function are unnecessary.

Reply:
Yeah, flatpak_fail_error is meant to be called with NULL (no error reporting) or null-initialized GError *. So this is not needed.